stdlib/public/runtime/MutexWin32.cpp

```diff
@@ -0,0 +1,90 @@
+//===--- Mutex.cpp - ------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Mutex, ConditionVariable, Read/Write lock, and Scoped lock implementations
+// using Windows Slim Reader/Writer Locks and Conditional Variables.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Mutex.h"
+
+using namespace swift;
+
+void ConditionPlatformHelper::init(CONDITION_VARIABLE &condition) {
+  InitializeConditionVariable(&condition);
+}
+
+void ConditionPlatformHelper::destroy(CONDITION_VARIABLE &condition) {
+}
+
+void ConditionPlatformHelper::notifyOne(CONDITION_VARIABLE &condition) {
+  WakeConditionVariable(&condition);
+}
+
+void ConditionPlatformHelper::notifyAll(CONDITION_VARIABLE &condition) {
+  WakeAllConditionVariable(&condition);
+}
+
+void ConditionPlatformHelper::wait(CONDITION_VARIABLE &condition,
+								   SRWLOCK &mutex) {
+  SleepConditionVariableSRW(&condition, &mutex, INFINITE, 0);
```

Comment:
Looks like this is one of the few functions that return an error...

> If the function succeeds, the return value is nonzero.
> If the function fails, the return value is zero. To get extended error information, call GetLastError.

...consider testing for an error and calling fatalError like is done in MutexPThread.cpp.